pkg/util/deployment/deployment.go

```diff
@@ -67,14 +79,24 @@ func GetOldRCs(deployment extensions.Deployment, c client.Interface) ([]*api.Rep
 // Returns an RC that matches the intent of the given deployment.
 // Returns nil if the new RC doesnt exist yet.
 func GetNewRC(deployment extensions.Deployment, c client.Interface) (*api.ReplicationController, error) {
+	return GetNewRCCustomized(deployment, c,
+		func(namespace string) ([]api.ReplicationController, error) {
+			rcList, err := c.ReplicationControllers(namespace).List(labels.Everything(), fields.Everything())
```

Comment:
How about adding the podTemplateHash label into the replication controller and filter based on it instead of fetching all the rcs?